packages/terra-date-picker/src/react-datepicker/calendar.jsx

```diff
@@ -361,18 +388,26 @@ export default class Calendar extends React.Component {
 
   localizeDate = date => localizeDate(date, this.props.locale)
 
-  increaseMonth = () => {
+  increaseMonth = (event) => {
     this.setState({
-      date: addMonths(cloneDate(this.state.date), 1)
+      isMonthChanged: true,
+      date: addMonths(cloneDate(this.state.date), 1).startOf('month')
     }, () => this.handleMonthChange(this.state.date))
-    this.props.setPreSelection(addMonths(cloneDate(this.state.date), 1));
+    this.props.setPreSelection(addMonths(cloneDate(this.state.date), 1).startOf('month'));
+    if(event.target.type === undefined) {
+      this.setState({ calendarIsKeyboardFocused : false});
+    }
   }
 
-  decreaseMonth = () => {
+  decreaseMonth = (event) => {
     this.setState({
-      date: subtractMonths(cloneDate(this.state.date), 1)
+      isMonthChanged: true,
+      date: subtractMonths(cloneDate(this.state.date), 1).startOf('month')
     }, () => this.handleMonthChange(this.state.date))
-    this.props.setPreSelection(subtractMonths(cloneDate(this.state.date), 1));
+    this.props.setPreSelection(subtractMonths(cloneDate(this.state.date), 1).startOf('month'));
+    if(event.target.type === undefined) {
```

Comment:
Is this check for determining whether the button was clicked or hitting the Enter key? Can you add a brief comment what this does as it wasn't obvious at first glance? Same in `increaseMonth`.

Reply:
Added c436d1a